core/src/main/scala/com/banno/kafka/RecordStream.scala

```diff
@@ -149,13 +149,15 @@ object RecordStream {
   type Incoming[F[_], K, V] = RecordStream[F, IncomingRecord[K, V]]
   type Batched[F[_], A] = RecordStream[F, IncomingRecords[A]]
 
-  private abstract class Impl[F[_]: Applicative, A](
+  private abstract class Impl[F[_]: Temporal, A](
       val consumer: ConsumerApi[F, GenericRecord, GenericRecord]
   ) extends RecordStream[F, A] {
     def readProcessCommit[B](process: A => F[B]): Stream[F, B] =
-      records.evalMap { x =>
-        process(x) <* consumer.commitSync(nextOffsets(x))
-      }
+      consumer.readProcessCommitAux[A, B](keepAliveInterval)(
+        records,
+        process,
+        a => nextOffsets(a).view.mapValues(_.offset - 1).toMap,
```

Comment:
Forgive my ignorance, but why the `-1` here?

Reply:
See elsewhere in `RecordStream` ... it looks like `nextOffsets` already computes the incremented value, but the method we're calling in `ConsumerApi` is expected the unincremented value, and increments it itself.